staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -174,12 +174,17 @@ func getIntOrPercentValue(intOrStr *IntOrString) (int, bool, error) {
 	case Int:
 		return intOrStr.IntValue(), false, nil
 	case String:
```

Comment:
reviewing call sites for safety:

```
getIntOrPercentValue
	GetValueFromIntOrPercent
		DaemonSetsController#getUnavailableNumbers
		NewRSNewReplicas(deployment)
		ResolveFenceposts
			MaxUnavailable(deployment apps.Deployment)
			MaxSurge(deployment apps.Deployment)
		DisruptionController#getExpectedPodCount(pdb)
	ValidateRollingUpdateDeployment
	ValidateRollingUpdateDaemonSet
```

called on these API fields:
* `DaemonSet.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable`

    String values are guaranteed to match `^[0-9]+%$` by `ValidatePositiveIntOrPercent`: https://github.com/kubernetes/kubernetes/blob/ab3ed8c19cf3fde87407ccacbc68939421585346/pkg/apis/apps/validation/validation.go#L344-L347

* `Deployment.Spec.Strategy.RollingUpdate.MaxSurge`
* `Deployment.Spec.Strategy.RollingUpdate.MaxUnavailable`

    String values are guaranteed to match `^[0-9]+%$` by `ValidatePositiveIntOrPercent`: https://github.com/kubernetes/kubernetes/blob/ab3ed8c19cf3fde87407ccacbc68939421585346/pkg/apis/apps/validation/validation.go#L432-L436

* `PodDisruptionBudget.Spec.MaxUnavailable`
* `PodDisruptionBudget.Spec.MaxAvailable`

    String values are guaranteed to match `^[0-9]+%$` by `ValidatePositiveIntOrPercent`: https://github.com/kubernetes/kubernetes/blob/ab3ed8c19cf3fde87407ccacbc68939421585346/pkg/apis/policy/validation/validation.go#L47-L64